nanoFirmwareFlasher.Library/FileDeployment/FileDeploymentManager.cs

```diff
@@ -0,0 +1,244 @@
+﻿using nanoFramework.Tools.Debugger;
+using nanoFramework.Tools.Debugger.Extensions;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace nanoFramework.Tools.FirmwareFlasher.FileDeployment
+{
+    public class FileDeploymentManager
+    {
+        private FileDeploymentConfiguration _configuration;
+        private VerbosityLevel _verbosity;
+        private string _serialPort;
+
+        public FileDeploymentManager(string configFilePath, string originalPort, VerbosityLevel verbosity)
+        {
+            _configuration = JsonConvert.DeserializeObject<FileDeploymentConfiguration>(File.ReadAllText(configFilePath));
```

Comment:
Shouldn't this have a sanity check for malformed json?
Maybe it can't even be deserialize, or the properties are misspelled...

Reply:
if this one fails, it gets trapped by the caller and has a specific error message.

Reply:
Perfect! Didn't look there.